src/method.jl

```diff
@@ -280,30 +322,58 @@ function SolverCore.solve!(
           iter_type,
         ],
       )
+
+    set_status!(
+      stats,
+      get_status(
+        nlp,
+        elapsed_time = stats.elapsed_time,
+        iter = stats.iter,
+        optimal = solved,
+        infeasible = infeasible,
+        penalty_too_large = penalty_too_large,
+        max_eval = max_eval,
+        max_time = max_time,
+        max_iter = max_iter,
+      ),
+    )
+
+    callback(nlp, solver, stats)
+
+    done = stats.status != :unknown
   end
 
-  if solved
-    status = :first_order
+  set_solution!(stats, x)
+  stats
+end
+
+function get_status(
```

Comment:
Isn't this function being repeated in several packages? Should it go into SolverCore?

Reply:
Yes, I think there are small differences though. But making a unified `get_status` in SolverCore would be great.
It is probably simpler if I make an issue in SolverCore to handle this separately.